app/routes/__index/domains/index.tsx

```diff
@@ -37,10 +81,10 @@ export default function DomainsIndexRoute() {
         navigate(domain.id.toString());
         break;
       case 'DELETE':
-        submit({ id: domain.id.toString() }, { method: 'delete' });
+        submit({ id: domain.id.toString(), intent: 'delete-record' }, { method: 'delete' });
```

Comment:
Question: can you not put the `id` in a `<Input type="hidden" />` in the form?  That's the more common way to store it.